include/bluetooth/gatt_dm.h

```diff
@@ -302,7 +302,13 @@ int bt_gatt_dm_data_release(struct bt_gatt_dm *dm);
  *
  * @param[in] dm Discovery Manager instance
  */
+#ifdef CONFIG_BT_GATT_DM_DATA_PRINT
 void bt_gatt_dm_data_print(const struct bt_gatt_dm *dm);
+#else
+static inline void bt_gatt_dm_data_print(const struct bt_gatt_dm *dm)
```

Comment:
why is it necessary to have an empty function?

Could you simply set BT_GATT_DM_DATA_PRINT=y for samples that use bt_gatt_dm_data_print?

Reply:
If this option is disabled code should still compile. This is API that not only sample use - we do for instance. If option is disabled data will not be printed - what's wrong with it? It is the same practice as with any kind of debug/print/log function.